### ISSUE TITLE
Repeat require statement in subclasses

### ISSUE THREAD

**Issue by plorent** [gigafied/minion#5]
If I define a Class like

``` javascript
minion.define("core.facade", {

    require : [
        "core.control.Controller",
        "core.model.Model",
        "core.view.View",
    ],

    Facade : minion.extend("minion.Singleton", {
})
```

and then create  a subclass of Facade, I have to repeat the require statement:

``` javascript
minion.define("application.facade", {

    require : [
        "core.control.Controller",
        "core.model.Model",
        "core.view.View",
    ],

    ApplicationFacade : minion.extend("core.facade.Facade", {
})
```

Is this correct? My expectation was that I wouldn't have to repeat the require statement.

Peter

**Comment by gigafied** [comment]
The only time you have to repeat the require statement is if you use those Classes directly in ApplicationFacade.

If you're familiar with Java or AS3, think of the requires array as import statements. If I:

`import core.control.Controller` in `Facade`, if I want to use the `Controller` class in `ApplicationFacade` as well, I have to import it there too.

If you don't use the classes "imported" in the sub class, then you should not have to include them in the `require` array.

However, I just tested this and it appears there is a bug with `this.sup()` that will not use the right `__imports` object when running super methods.

I'll look into fixing this tomorrow, but as it stands right now, you may have to list some dependencies twice (though this is not the intended behavior.)

**Comment by plorent** [comment]
I have run into stack overflow issues as well with the require statement. I've refactored my code to circumvent the problem but if you want me to keep track of these issues and post here, let me know.

**Comment by gigafied** [comment]
That would be great, I know I've run into issues with circular dependencies, but any other info you can provide would be helpful.

**Comment by gigafied** [comment]
This is now fixed and pushed in 1.5.5, I've also published to npm.

**Comment by plorent** [comment]
Awesome. I'll check it out. Thanks.
